src/shogun/features/RandomKitchenSinksDotFeatures.cpp

```diff
@@ -35,20 +36,15 @@ RandomKitchenSinksDotFeatures::RandomKitchenSinksDotFeatures(
 
 SGMatrix<float64_t> RandomKitchenSinksDotFeatures::generate_random_coefficients()
 {
-	SGVector<float64_t> vec = generate_random_parameter_vector();
-	SGMatrix<float64_t> random_params(vec.vlen, num_samples);
-	for (index_t dim=0; dim<random_params.num_rows; dim++)
-		random_params(dim, 0) = vec[dim];
+	SGMatrix<float64_t> random_params(feats->get_dim_feature_space() + 1 , num_samples);
```

Comment:
as explained before, this feature space dim +1 shouldnt be harcoded in here. You need to add a purely virtual method to the base class that returns the number of dimensions. OR you first ask for the first parameter vector, then allocate the matrix and fill the rest. This is a general class that could work with other parametrizations so we don't want to fixate this here.

Reply:
Question,  the random_params matrix transform our dimensionality of feature space from n -> k. So, shouldn't it's rows always be feature space dim  + 1 ?.

Reply:
no, *as explained above* that is only the case if you use RFF features with the offset structure that is used here. But that can change for different random feature classes, maybe some RFF classes dont even use offsets. Rule of thumb: never hardcode such things, always put them into a method. That is the whole point of doing this in an object oriented design.